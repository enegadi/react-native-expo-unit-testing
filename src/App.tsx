import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import CustomTextInput from "./components/CustomTextInput";
import React from "react";

export default function App() {
  return (
    <View style={styles.container}>
      <Text>Open up App.tsx to start working on your app!</Text>
      <StatusBar style="auto" />
      <CustomTextInput
        placeholder="Enter text"
        onChangeText={(text: string) => console.log(text)}
        value=""
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
