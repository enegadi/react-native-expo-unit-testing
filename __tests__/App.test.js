import React from "react";
import renderer from "react-test-renderer";
import App from "../src/App";

describe("<App />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("has 1 child", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children.length).toBe(2);
  });

  it("has a specific child component", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree.children[0].type).toBe("Text");
  });
});
