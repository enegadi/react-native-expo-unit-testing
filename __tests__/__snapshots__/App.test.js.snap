// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Open up App.tsx to start working on your app!
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Enter text"
    style={
      {
        "backgroundColor": "white",
        "borderColor": "blue",
        "borderRadius": 10,
        "borderWidth": 2,
        "color": "black",
        "elevation": 3,
        "fontSize": 16,
        "height": 40,
        "paddingLeft": 10,
        "paddingRight": 10,
        "shadowColor": "gray",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 4,
      }
    }
    value=""
  />
</View>
`;
